### PR TITLE
Add autoscale test for ingester

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

**Comment by kevinearls** [comment]
@jpkrohling @objectiser @rubenvp8510 Please review

File: test/e2e/autoscale_test.go [hunk 1]
```diff
@@ -129,8 +157,73 @@ func (suite *AutoscaleTestSuite) TestAutoScaleCollector() {
...
+			},
+		},
+	}
+
+	return jaeger
+}
+
 // Create a simple-prod instance with optional values for autoscaling the collector
-func getSimpleProd(name, namespace, cpuResourceLimit, memoryResourceLimit string, autoscaleCollector bool) *v1.Jaeger {
+func getSimpleProd(name, namespace, cpuResourceLimit, memoryResourceLimit string, autoscaleCollector bool) *v1.Jaeger { // FIXME remove autoscaleCOlletor option
```
Comment (rubenvp8510) [inline test/e2e/autoscale_test.go:226]: I think for the case of the ingester we are not interesting on having an auto-scaling for collector, but a fixed number of collector replicas. Not sure if this makes the test more or less reliable though .

File: test/e2e/autoscale_test.go [hunk 2]
```diff
@@ -129,8 +157,73 @@ func (suite *AutoscaleTestSuite) TestAutoScaleCollector() {
 	require.Greater(t, maxCollectorCount, 1, "Collector never scaled")
 }
 
+func getSimpleStreaming(name, namespace string) *v1.Jaeger {
+	kafkaClusterURL := fmt.Sprintf("my-cluster-kafka-brokers.%s:9092", kafkaNamespace) // FIXME this may need to change
```
Comment (jpkrohling) [inline test/e2e/autoscale_test.go:161]: Why would this need to be changed?

File: test/e2e/autoscale_test.go [hunk 3]
```diff
@@ -129,8 +157,73 @@ func (suite *AutoscaleTestSuite) TestAutoScaleCollector() {
...
+							corev1.ResourceCPU:    resource.MustParse(cpuResourceLimit), // FIXME this might need to change
+							corev1.ResourceMemory: resource.MustParse(memoryResourceLimit),
+						},
+					},
+				},
+				Options: v1.NewOptions(map[string]interface{}{
+					"kafka.consumer.topic":      "jaeger-spans",
+					"kafka.consumer.brokers":    kafkaClusterURL,
+					"ingester.deadlockInterval": "0",
+					"ingester.parallelism":      "6900",
```
Comment (jpkrohling) [inline test/e2e/autoscale_test.go:201]: That's a very specific number :-) What's the story behind it?

File: test/e2e/autoscale_test.go [hunk 4]
```diff
@@ -129,8 +157,73 @@ func (suite *AutoscaleTestSuite) TestAutoScaleCollector() {
...
+					MaxReplicas: &maxReplicas,
+				},
+			},
+			Storage: v1.JaegerStorageSpec{
+				Type: "elasticsearch",
+				Options: v1.NewOptions(map[string]interface{}{
+					"es.server-urls": esServerUrls,
+				}),
+			},
+			Ingress: v1.JaegerIngressSpec{ // FIXME do we need this?
```
Comment (jpkrohling) [inline test/e2e/autoscale_test.go:215]: You don't need this, if all you are doing is checking the number of replicas.

File: test/e2e/autoscale_test.go (hunk 2 above)
Comment (objectiser) [inline test/e2e/autoscale_test.go:161]: FIXME comment?

File: test/e2e/autoscale_test.go (hunk 4 above)
Comment (rubenvp8510) [reply test/e2e/autoscale_test.go:215]: It's true, this is not needed.

File: test/e2e/autoscale_test.go (hunk 1 above)
Comment (objectiser) [reply test/e2e/autoscale_test.go:226]: @kevinearls another  FIXME? Should be solved.

File: test/e2e/autoscale_test.go (hunk 3 above)
Comment (rubenvp8510) [reply test/e2e/autoscale_test.go:201]: Is an attempt to saturate the resources of the ingester pod, I saw that is easier if we have high parallelism.

**Review by objectiser — COMMENTED** [review]
@kevinearls Has it been tested on OCP?

File: test/e2e/autoscale_test.go (hunk 1 above)
Comment (kevinearls) [reply test/e2e/autoscale_test.go:226]: I'll clean this up.  Originally I thought I could use the same method to create the CR for the collector and ingester test, but that would have ended up being more work than it was worth.

File: test/e2e/autoscale_test.go (hunk 2 above)
Comment (kevinearls) [reply test/e2e/autoscale_test.go:161]: It doesn't, I thought I had taken out all TODOs and FIXME.  Originally I was thinking of creating a kafka instance programatically in the same namespace as the test, but using the kafka instance that we already create works fine and is less work.  So I will clean this up.

File: test/e2e/autoscale_test.go (hunk 3 above)
Comment (kevinearls) [reply test/e2e/autoscale_test.go:201]: You'll have to ask @rubenvp8510

File: test/e2e/autoscale_test.go (hunk 2 above)
Comment (kevinearls) [reply test/e2e/autoscale_test.go:161]: Sorry...I thought I had removed them all....

**Comment by kevinearls** [comment]
@objectiser  Yes, it has been tested on OCP

File: test/e2e/autoscale_test.go (hunk 3 above)
Comment (jpkrohling) [reply test/e2e/autoscale_test.go:200]: > Is an attempt to saturate the resources of the ingester pod, I saw that is easier if we have high parallelism.

Sounds reasonable, but why "6900" and not, say, "7000", or even "10000" :-)

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**